### ISSUE TITLE
Builds are sometimes not added to the history list

### ISSUE THREAD

**Issue by dominikgolda** [Soloplan/whatson#31]
![buildsSometimesNotAdded](https://user-images.githubusercontent.com/7096402/72882660-4597e480-3d03-11ea-9668-1d7a2e83c968.png)

There is a failing unit test BuildsShouldNotBeMissedWhenBuildsAreRunningOneAfterAnother. I think it fails on some mocks, which were not refactored, but maybe fixing it is some starting point.

**Comment by dominikgolda** [comment]
af387d0b is likely cause of the problem. It certainly breaks the unit test.

**Comment by dominikgolda** [comment]
The bug itself is fixed. Only the unit test remains to be fixed.

**Comment by dominikgolda** [comment]
It turns out the unit test was failing because the feature wasn't working. The problem was due to how adding to queue works. We first remove the last value, then we add new one and then we sort by age. As a result when we had ex. builds 10, 9, 8, 7, 6 and we added 11, 10, 9, 8, 7 in turns the result was that we ended up with 11, 10, 10, 9, 7. The steps were:
- Adding 11:
11, 10, 9, 8, 7
-Adding 10:
11, 10, 10, 9, 8
- Adding 9:
11, 10, 10, 9, 9
-Adding 8:
11, 10, 10, 9, 8
-Adding 7:
11, 10, 10, 9, 7

The fix was to just query the missing build and add it to queue, which should improve performance. Alternatively we could clear the history before reloading it.

Apart from what was actually not working the unit test required adjusting to changes in JenkinAPI class.
